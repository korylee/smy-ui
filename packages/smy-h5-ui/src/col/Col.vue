<template>
  <div
    :class="{
      [`smy-col--span-${span}`]: span,
      [`smy-col--offset-${offset}`]: offset,
    }"
    :style="style"
    class="smy-col"
    v-on="$listeners"
  >
    <slot />
  </div>
</template>
<script>
import { createChildrenMixin } from '../_mixins/relation'
import { convertToUnit } from '../_utils/shared'
import { props } from './props'

export default {
  name: 'SmyCol',
  mixins: [createChildrenMixin('row', { children: 'cols' })],
  props,

  data: () => ({
    style: {
      paddingLeft: 0,
      paddingRight: 0,
    },
  }),

  methods: {
    setPadding({ left, right }) {
      this.style = {
        paddingLeft: convertToUnit(left),
        paddingRight: convertToUnit(right),
      }
    },
  },
}
</script>

<style lang="less">
@import '../_styles/common.less';
@import './col.less';
</style>
