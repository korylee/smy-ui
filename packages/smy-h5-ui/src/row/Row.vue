<template>
  <div :style="style" class="smy-row smy--flex smy--box" v-on="$listeners">
    <slot />
  </div>
</template>
<script>
import { createParentMixin } from '../_mixins/relation'
import { toPxNum } from '../_utils/shared'
import { props } from './props'

export default {
  name: 'SmyRow',
  mixins: [createParentMixin('row', { children: 'cols' })],
  props,
  computed: {
    average() {
      return toPxNum(this.gutter) / 2
    },
    computePaddingWaterDispatcher() {
      return [this.gutter, this.cols]
    },
    style() {
      const { average } = this
      return {
        justifyContent: this.justify,
        alignItems: this.align,
        margin: average ? `0 -${average}px` : undefined,
        flexWrap: this.wrap,
      }
    },
  },

  watch: {
    computePaddingWaterDispatcher() {
      this.$nextTick(() => {
        const { average } = this
        this.cols.forEach((col) => {
          col.setPadding({ left: average, right: average })
        })
      })
    },
  },
}
</script>

<style lang="less">
@import '../_styles/common.less';
@import './row.less';
</style>
