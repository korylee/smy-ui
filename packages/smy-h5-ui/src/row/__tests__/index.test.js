import { mount } from '@vue/test-utils'
import Vue from 'vue'
import Row from '..'
import Col from '../../col'
import example from '../example'

test('test row example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test row & col plugin', () => {
  Vue.use(Row).use(Col)
  expect(Vue.component(Row.name)).toBeTruthy()
  expect(Vue.component(Col.name)).toBeTruthy()
})

test('test row flex', async () => {
  const wrapper = mount(Row)
  await Vue.nextTick()

  await wrapper.setProps({ justify: 'flex-start' })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({ justify: 'flex-end' })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({ justify: 'center' })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({ justify: 'space-between' })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({ justify: 'space-around' })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({ align: 'flex-start' })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({ align: 'flex-end' })
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setProps({ align: 'center' })
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test row click', async () => {
  const onClick = jest.fn()
  const wrapper = mount(Row, {
    listeners: { click: onClick },
  })

  await wrapper.trigger('click')
  expect(onClick).toHaveBeenCalledTimes(1)

  await wrapper.find('.smy-row').trigger('click')
  expect(onClick).toHaveBeenCalledTimes(2)

  wrapper.destroy()
})

test('test col in row', async () => {
  const template = `
  <smy-row>
    <smy-col :span="span" :offset="offset">1</smy-col>
    <smy-col :span="12">2</smy-col>
  </smy-row>
  `
  const wrapper = mount({
    data: () => ({
      span: 8,
      offset: 4,
    }),
    components: { SmyCol: Col, SmyRow: Row },
    template,
  })
  expect(wrapper.html()).toMatchSnapshot()
  await wrapper.setData({ span: 12, offset: 0 })
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})
