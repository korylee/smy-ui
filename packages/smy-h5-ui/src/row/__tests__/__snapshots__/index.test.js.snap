// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test col in row 1`] = `
"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap;">
  <div class="smy-col smy-col--span-8 smy-col--offset-4" style="padding-left: 0px; padding-right: 0px;">1</div>
  <div class="smy-col smy-col--span-12" style="padding-left: 0px; padding-right: 0px;">2</div>
</div>"
`;

exports[`test col in row 2`] = `
"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap;">
  <div class="smy-col smy-col--span-12" style="padding-left: 0px; padding-right: 0px;">1</div>
  <div class="smy-col smy-col--span-12" style="padding-left: 0px; padding-right: 0px;">2</div>
</div>"
`;

exports[`test row example 1`] = `
"<div class="layout-example">
  <div class="app-demo-title">栅格系统</div>
  <div class="demo-item">
    <div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap;">
      <div class="flex-item smy-col smy-col--span-24" style="padding-left: 0px; padding-right: 0px;">span:24</div>
    </div>
    <div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap;">
      <div class="flex-item smy-col smy-col--span-12" style="padding-left: 0px; padding-right: 0px;">span:12</div>
      <div class="flex-item smy-col smy-col--span-12" style="padding-left: 0px; padding-right: 0px;">span:12</div>
    </div>
    <div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap;">
      <div class="flex-item smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">span:8</div>
      <div class="flex-item flex-item--light smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">span:8</div>
      <div class="flex-item smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">span:8</div>
    </div>
    <div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap;">
      <div class="flex-item smy-col smy-col--span-6" style="padding-left: 0px; padding-right: 0px;">span:6</div>
      <div class="flex-item flex-item--light smy-col smy-col--span-6" style="padding-left: 0px; padding-right: 0px;">span:6</div>
      <div class="flex-item smy-col smy-col--span-6" style="padding-left: 0px; padding-right: 0px;">span:6</div>
      <div class="flex-item smy-col smy-col--span-6" style="padding-left: 0px; padding-right: 0px;">span:6</div>
    </div>
  </div>
  <div class="app-demo-title">偏移</div>
  <div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: wrap;">
    <div class="smy-col smy-col--span-16 smy-col--offset-8" style="padding-left: 0px; padding-right: 0px;">
      <div class="flex-item">span:8</div>
    </div>
    <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
      <div class="flex-item flex-item--light">span:8</div>
    </div>
    <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
      <div class="flex-item">span:8</div>
    </div>
  </div>
  <div class="app-demo-title">对齐</div>
  <div class="demo-item">
    <div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap;">
      <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
        <div class="flex-item">span:8</div>
      </div>
      <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
        <div class="flex-item">span:8</div>
      </div>
    </div>
    <div class="smy-row smy--flex smy--box" style="justify-content: center; align-items: flex-start; flex-wrap: nowrap;">
      <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
        <div class="flex-item">span:8</div>
      </div>
      <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
        <div class="flex-item">span:8</div>
      </div>
    </div>
  </div>
  <div class="app-demo-title">分栏间隔</div>
  <div class="smy-row smy--flex smy--box" style="align-items: flex-start; margin: 0px -5px; flex-wrap: nowrap;">
    <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
      <div class="flex-item">span:8</div>
    </div>
    <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
      <div class="flex-item flex-item--light">span:8</div>
    </div>
    <div class="smy-col smy-col--span-8" style="padding-left: 0px; padding-right: 0px;">
      <div class="flex-item">span:8</div>
    </div>
  </div>
</div>"
`;

exports[`test row flex 1`] = `"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap; justify-content: flex-start;"></div>"`;

exports[`test row flex 2`] = `"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap; justify-content: flex-end;"></div>"`;

exports[`test row flex 3`] = `"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap; justify-content: center;"></div>"`;

exports[`test row flex 4`] = `"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap; justify-content: space-between;"></div>"`;

exports[`test row flex 5`] = `"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap; justify-content: space-around;"></div>"`;

exports[`test row flex 6`] = `"<div class="smy-row smy--flex smy--box" style="align-items: flex-start; flex-wrap: nowrap; justify-content: space-around;"></div>"`;

exports[`test row flex 7`] = `"<div class="smy-row smy--flex smy--box" style="align-items: flex-end; flex-wrap: nowrap; justify-content: space-around;"></div>"`;

exports[`test row flex 8`] = `"<div class="smy-row smy--flex smy--box" style="align-items: center; flex-wrap: nowrap; justify-content: space-around;"></div>"`;
